### ISSUE TITLE
Improve "evaluation box"

### ISSUE THREAD

**Issue by nitedog** [w3c/wai-atag-report-tool#60]
In addition to the "Your Evaluation" page, there is also a "Your evaluation" sidebar/alert. This is in itself confusing (see issue #59 for more on this topic). This issues is about the "Your evaluation" sidebar/alert alone.

On my configuration, I have to switch to full screen for it to appear on the left. It otherwise appears at the top on every single page and it is very unclear what it does. I finally pressed "new evaluation" and now it constantly just says "Set up information about your evaluation" on every single page but does not indicate how to do that setup, nor does it provide a way to remove that quite intrusive alert.

I'm not sure what the solution is but feel that there is little use of that sidebar/alert at all.

**Comment by hidde** [comment]
I'll respond to the placement and confusion issues separately below.

## Placement

The breakpoint for this “evaluation box” to appear on the left is 960px in most browsers (defined as `60em`, so depends on browser default for `1em`). [According to Statcounter](https://gs.statcounter.com/screen-resolution-stats/desktop/worldwide) a large majority of desktop web users use something larger than that. 

This breakpoint is the same one that is used by the main WAI website and it is the ideal breakpoint given space available and content we want to display.

## Confusing

Thanks for the feedback on how you experienced it. The goal of this widget is the opposite of what you are experiencing: to remove confusion and make it clear what you are working on and how it is going. It is meant to help you, not confuse you. It seems that is not going so well.

I am working on improving what is displayed in this box in the various states. The most useful state is visible when you have evaluated one or more criteria, because then it displays how many you have worked on.

<img width="314" alt="" src="https://user-images.githubusercontent.com/178782/79770949-5f5ebd00-832e-11ea-94c3-c5f62c6a8456.png">

**Comment by hidde** [comment]
I have updated this to be less dynamic, and with that, hopefully less confusing. It is now based on which page it appears on, rather than the complex combination of factors it was based on before. 

- Start page: it welcomes the user and urges to start evaluation or import
- Your evaluation: it explains what this page is for
- Evaluating principle 1-8: it shows how many success criteria were reported

**Comment by nitedog** [comment]
I personally still find it confusing. Here situation:

1. I'm a first-time user coming to this tool -- I'm on the overview/start page and should probably read some the content here
2. The first thing that comes to my attention is the navigation menu -- good, I see that I'm at "Start" and have some orientation
3. The second thing that comes to my attention is "Your evaluation" box with a prominent "Start new evaluation" link/button
4. I'm very tempted to select that link/button but it then takes me away from the content that I should be reading first
5. I land on "Your Evaluation" page with yet another "Your evaluation" box and many form fields, and not sure what to do

I feel this alert/sidebar is too intrusive and redirects people wrongly.

**Comment by hidde** [comment]
More input on this [in the second survey](https://www.w3.org/2002/09/wbs/35532/acert-2/results#xq8)

**Comment by hidde** [comment]
For context, from the earlier survey:

## Background:

* Users of this tool will be doing multiple evaluations, maybe in the same week, especially so if they are a professional accessibility evaluator
* We store an evaluation locally on the user's computer (as requested in #26). We think this is useful, but realise it could be confusing.<br><i>Note: it is listed in the Note on the front page, but we can assume not everyone will read</i>
* It is generally useful for evaluations to have names (and this was a comment in the previous survey)
* People might want to go back into an evaluation they did previously, for re-evaluation or maybe to adjust a comment or result

## User need 

Following this, the user need is: as an accessibility evaluator, I want to have easy access to what my current evaluation is, so that I can feel confident I am in the correct evaluation document.

## How ‘evaluation box’ tries to address this user need

Suggested new element to help with this: the evaluation box.

* It is dynamic. If you are new into the tool, it suggests to start a new evaluation or import. if you have a locally saved version, it displays the name of that version and has option to Save or Clear. (In WCAG EM Reprt Tool, all those options are always available; this makes them contextually available).
* It displays in real time how many success criteria you've evaluated ('evaluated' means you have selected some option for that SC).
* On desktop it floats on the left, on mobile it is on the top of the page.
* It is meant to give users confidence about which evaluation they are currently working on (probably always helpful, but especially helpful when doing evaluations for multiple clients/projects). This is similar functionality to Google Docs displaying name of doc prominently.
* It is meant to help people understand if they're in a previously saved evaluation versus a new one.

**Comment by nitedog** [comment]
I agree with the above bullets except for this one, which is not related to a user need IMO:

> - On desktop it floats on the left, on mobile it is on the top of the page

For screen readers and everyone not using full-width browser windows this always appears first. My issues is really the placement that seems too prominent to me.

**Comment by hidde** [comment]
Thanks for clarifying! The 'always visible' is meant to be helpful (but it will only be if the content of it is helpful).

I can experiment with other positions for this information too. And I could add it later in source order to make the experience better in a screenreader.

Here's a mockup of placement at the bottom. Less prominent, but I feel it's a bit weirder:

<img width="1295" alt="" src="https://user-images.githubusercontent.com/178782/80011339-07f65380-84cc-11ea-82f8-700dd484b286.png">

**Comment by nitedog** [comment]
To me, this area on the left represents some sort of navigation, which this is not. Can it be on the right (ie. visually and also non-visually *after* the main content)?

**Comment by bakkenb** [comment]
Bottom location does not work for me, makes it even more confusing. I need some time to think through this one and really consider/ponder what is being conveyed in this box on each screen and if there is a more simple way to do it.

**Comment by slhenry** [comment]
[EOWG 24 April discussion](https://www.w3.org/2020/04/24-eo-minutes.html#item03)

**Comment by slhenry** [comment]
**_Overview:_** I agree the info is useful. I think it should be provided elsewhere, rather than in this box.

**_Some more input._**

Situation: “Users of this tool will be doing multiple evaluations, maybe in the same week, especially so if they are a professional accessibility evaluator”… “as an accessibility evaluator, I want to have easy access to what my current evaluation is, so that I can feel confident I am in the correct evaluation document”

SLH comment: I’m “on the fence” on the importance of this. It seems like most people will be doing a report on their own tool, and fewer will be doing it on multiple authoring tools around the same time. And those doing reports for multiple tools, will usually focus on one at a time. Even so, it would be good to be clear which tool the current report is for. I just think it can be quiet / unobtrusive.

A brainstorm would be to put where this is:
> ATAG Report Tool (Principle 1/of 8)

this instead:
> “ATAG Report for [tool name]

<hr>

> “It displays in real time how many success criteria you've evaluated”

I think generally a good thing to provide. Yet, I don’t think needs to have lots of screen real estate and focus. Should be tangentially available, rather than interrupt flow and visual hierarchy.

Brainstorms:

In the 24 April meeting, I *wondered* about adding “X of X” under each A/B in the nav. Possibly that might be helpful. More likely, it will be way too much clutter.

Next idea (which might be equally bad, or might be brilliant ;-):
What about designing so that the final view provides such information -- even more specifically – and guiding user to use that page to get their overview of what they’ve done and what they have left to do?

If the last page has a list something like:
* Passed: A.3.2.3, A 3.2.3, …
* Failed: A.3.6.1, A.3.6.2
* Cannot tell: A.4.2.1, A.4.2.2, …
* Not applicable: B.2.4.2, B.2.4.3, …
* Not Checked: B.4.2.1, B.4.2.1, …
(as suggested [https://github.com/w3c/wai-atag-report-tool/issues/94 here])

Then, people using the tool can see what is left to do. Maybe could include the “X of X” in this view (though not in the downloaded report).

**Comment by hidde** [comment]
> Situation: “Users of this tool will be doing multiple evaluations, maybe in the same week, especially so if they are a professional accessibility evaluator”… “as an accessibility evaluator, I want to have easy access to what my current evaluation is, so that I can feel confident I am in the correct evaluation document”
> 
> SLH comment: I’m “on the fence” on the importance of this. It seems like most people will be doing a report on their own tool, and fewer will be doing it on multiple authoring tools around the same time. And those doing reports for multiple tools, will usually focus on one at a time. Even so, it would be good to be clear which tool the current report is for. I just think it can be quiet / unobtrusive.

We don't really have data on how people do authoring tool evaluations, because very few people (if anyone at all) create ATAG evaluations.

For this project, I did talk to various people in our target audience who do WCAG evaluations, including independent accessibility consultants and accessibility consultants working for  accessibility consultancy firms. Multiple evaluations was a very common thing for them. 

I'd propose we extrapolate the presumed use case for WCAG evaluators to ATAG evaluators, because they are likely the same people. (Given the state of ATAG implementation, I think it is safe to say someone who can evaluate for it is a an experienced accessibility consultant).

I worry adding the tool name to the title  “ATAG Report for [tool name]” makes for a confusing interface, because we'd mix two things that are arguably functionally different. “ATAG Report Tool” has the function of being the name of the tool, “[tool name]” in the evaluation box addresses the user need of quickly identifying what the current evaluation is. I don't feel it can be as effective in identifying the current evaluation if it is in a functionally different thing.

> I think generally a good thing to provide. Yet, I don’t think needs to have lots of screen real estate and focus. Should be tangentially available, rather than interrupt flow and visual hierarchy.

I disagree about “lots of screen real estate and focus”. 

Let's consider it in terms of visual columns, which most of our users will encounter and determine by what's most important.

If you have just over 1000px available in the width, which is an overwhelming majority of desktop users:

* nav bar: 8 columns (all that are available)
* evaluation itself: 6 columns (a large chunk of the maximum)
* evaluation box: 2 columns (only a third of the maximum, i.e. a small part)
* footer: 8 columns + left and right fully used with background color

_Everything_ in this UI has more screen real estate and focus than this box.

Some users, a much smaller group,  have less `px` available in the width (including users of mobile phones, users with low resolution screens and/or higher zoom levels or certain tablets): everything will be full width. This is because there is not enough space to create columns (a physical constraint that we cannot change and do not change on the main WAI site either). We could mitigate this somewhat by placing it underneath the main area, but then it is harder to find, and not helping to quickly identify where they are. 

So, in summary: I do not agree flow or visual hierarchy are disrupted by this box. Hiding it in some other place goes beyond the point this box tries to solve.

> Next idea (which might be equally bad, or might be brilliant ;-):
> What about designing so that the final view provides such information -- even more specifically – and guiding user to use that page to get their overview of what they’ve done and what they have left to do?

I like this idea. Let me do a mockup in #94 and let's discuss it further there.

*** 

Proposal: let's leave this box in as it is, because it addresses the user need, it is helpful to some, the perceived problems apply only to a small minority and the alternatives are less attractive, because they make things less clear.

**Comment by slhenry** [comment]
"perceived problems" are significant for many low vision users. We need to be able to use the full browser width. In QuickRef, that is achieved with Hide & Show sidebar.

We need to come up with a good site-wide solution for this.

**Comment by hidde** [comment]
How wouldn't those users have exactly the same problems on our main website, which uses the same grid in the same way?

Or on any of the other websites that don't use the full width of the browser window (which is very uncommon for websites to do these days, I can only name two from the top of my head).

**Comment by iadawn** [comment]
Just looking through all this and considering the purpose of the box and the user need I thought I would through my 2 pence/cent/currency of choice in.

The user need is:

> as an accessibility evaluator, I want to have easy access to what my current evaluation is, so that I can feel confident I am in the correct evaluation document

The only element of the evaluation box that responds to this user need is the text "Evaluating ...". The other elements do the following, to my thinking:

- "Reported on x out of x success criteria" - provides an indication of progress. I think this is useful.
- "View report" - quick access to the report. I think this is useless as it is prominently in the navigation bar. As such this is replicating with a strong call to action which isn't the primary call to action on the page. 
- "Clear" - Resets the reporting tool. I think this is useless at best and dangerous at worse and should be removed completely and utterly. The text is unclear, the positioning allows for accidental clicks and the styling temptingly draws the eye. I know there is a confirmation box but they are well known for there uselessness.

Based on this I think the only useful things are the name of the tool and the progress. On that basis I think there is scope to style this differently and something more inline with what has been proposed by others:

- Remove the Your evaluation box
- Single column presentation using 6 grid columns starting in column 1 
- Include a page header with:
  - Report for "tool name"
  - Progress - could be a visual bar with the numeric representation presented as well

**Comment by hidde** [comment]
@iadawn Thanks for your input! 

I see you feel the buttons in the evaluation box are not useful. The reason I believe they make sense to exist where they do, is that they very closely related to what the current evaluation: they let the user look more closely or get rid of the current evaluation in the same place. 

If we are to remove these buttons from this box, I'd like us to have a location in mind for them to go that is better than where they currently are. 'View Report' is, as you say, available in the navigation, but for some users having it available under this button makes it easier to find, I don't feel it hurts to have multiple links to the same (important) place. The 'Clear' button is important, because if someone finds out they are looking at a given evaluation, but it is not the one they want to work on, they'd need an obvious way to clear it.

***

To clarify the suggestion for the grid, this is what the grid looks like currently. The first and last grid column are flexible and use the 'remaining space'.

<img width="1651" alt="Screenshot 2020-05-20 at 21 31 41" src="https://user-images.githubusercontent.com/178782/82489066-4db13680-9ae1-11ea-90d3-c2ab9c7c4c20.png">

> Single column presentation using 6 grid columns starting in column 1

There is various ways I can interpret this. In this screenshot I have marked an area (1) and an area (2). Does either of these correspond to what you would like to see as the one column presentation? 

What would go in that column? All of the content, or just the proposed page header? Would the navigation also be 6 columns wide (currently has 8 columns).

<img width="1101" alt="Screenshot 2020-05-20 at 21 35 25" src="https://user-images.githubusercontent.com/178782/82489364-d203b980-9ae1-11ea-9057-851f796e5d06.png">

**Comment by hidde** [comment]
I have done a bunch of updates to address some more needs raised in this issue ([ART main preview](https://atag-report-tool.hdv.now.sh/)).

* Moved evaluation box to the right of the screen (and in the markup, under the main content area), so that it has less focus
* Only display “Clear report” button if there is any data to clear
* Tweak wording so that we say “report” everywhere

**Comment by hidde** [comment]
@iadawn Thanks for sharing the Jakob Nielsen post. I hadn't realised before how it was relevant to our discussion, as our cancel button technically does more than clearing form content, but understand now that it is pretty much the same from a user perspective. 

I like the rename to “Start new report”, and see how clearing/canceling is implied in that.  Having it only on the Overview page also makes sense, as the use case (“someone sees report with existing data and doesn't want existing data”) is most likely to happen on that first page. 

Will update!

**Comment by slhenry** [comment]
_I discussed this with Hidde (which is sooo much nicer than typing text!), so brief notes here for others._

I certainly prefer it on the right/bottom rather than the left/top. I’m still not convinced that it’s good to have. I think we need to address one point [1]. Other than that, if others really want it, I’ll not say more against it.

[1] The one important thing that is in that box and no where else is the “Start new report”, which is currently on the first page. In smaller viewports and/or with large text/zoom, that is now at the very bottom of the page. That doesn’t seem right. It seems “Start new report” should be prominent on the Overview page &/or the first About/Start page. I think **this warrants re-thinking how people will use the tool, and when they might want to clear old info and start a new report**.
(Also note that I think it’s fine if they select that before they have entered any data and just want to get started -- even though it’s not required.)

>  Having it only on the Overview page also makes sense, as the use case (“someone sees report with existing data and doesn't want existing data”) is most likely to happen on that first page.

Actually, wouldn't it be on the second page where they have data? There's no data on the first page.

<hr>

As for the other info in that box:
* “View Report” is available as the last link in the main navigation (thus easy to see and get to), and is with every “Observations” box.
* “Reported on X out of XX success criteria.” is mildly useful. However, not as useful as the listing of which SCs have and have not yet been reported on, which is listed on the Report page.
* Name of Authoring Tool - I don’t think this is important. I don’t envision users going back and forth working on multiple ATAG evaluations at the same time. We don’t have it in the WCAG-EM Report Tool and no one has asked for it (although that doesn’t mean they wouldn’t have benefited from it).

While I don’t think they’re important, I also don’t think there’s anything wrong with the redundant info, as long as it doesn’t detract or degrade the user experience in other ways.

**Comment by slhenry** [comment]
side note: If this stays "Reported on X out of XX success criteria." then I think need to clarify the level if it's less than AAA.

Use case: I saw "Reported on 3 out of 24 success criteria." and thought: _That's not right, I though there were way more SCs then 24!_

maybe:
> Reported on X out of 24 Level A success criteria.
> Reported on X out of 39 Level A and AA success criteria.

**Comment by hidde** [comment]
As discussed in EOWG last Friday, we'd like the evaluation box to indicate better what's missing and what's not.

## Proposal

### What it is

* Add expand/collapse “details”
* Within it, list each principle, how many items are done 
* Displays progress bar, to (a) visually reinforce this is a display of progress, (b) show what that progress is for visually inclined users
* Contains percentage (visually hidden)
* Links to principle

<img width="302" alt="" src="https://user-images.githubusercontent.com/178782/84375605-20d3d900-abe0-11ea-9a03-89368261e3a0.png">

### How it helps users

* Can quickly find what's missing, for instance if they work on a report that has 59/60 done, this helps find the missing one

### How it does not help

* No info on which exact SCs are missing. I think this is fine, because one principle has less than 8 SCs, generally a lot less
* Link to principle does not look like traditional link. I think this is great, a visual aid in the form of a “chevron” is added.

**Comment by nitedog** [comment]
I really like this conceptually!

1. I don't think we need the expand/collapse inside the widget. I think the suggestion during the EOWG call was to have a show/hide of the entire widget, just like in the quickref (to collapse the entire widget to the right and make the content full-width, for those who prefer)

2. To me, the chevron looked to me like an expand/collapse icon. I thought that there could be further information hidden below at first. Maybe there can be a better visual cue?

3. I think the currently active page/step should not be clickable in the widget. I think it should have a slightly different visual presentation, like in the menu bar.

4. I think the text in this widget in the start/about page is not very useful. I think we should leave the same info here in the widget as we have on the other pages.

5. Should we an item in the widget that indicates whether start/about info has been provided? Seems also relevant for such an overview.

6. I don't feel too strongly but wonder if all fields/cells in this widget should be in just one column instead of currently two?

**Comment by hidde** [comment]
Thanks for the initial thoughts. Some responses:

> 2. To me, the chevron looked to me like an expand/collapse icon. I thought that there could be further information hidden below at first. Maybe there can be a better visual cue?

OK. On WAI sites, all our expand/collapse icons are +'es, we do not use chevrons.

Underline and chevrons are the most common link indicators I know of… if more people find the chevron confusing, we could consider these two versions:

<img width="655" alt="left: principle numbers with no chevron and no underline; right: principle numbers with no chevron but an underline added" src="https://user-images.githubusercontent.com/178782/84383995-998d6200-abed-11ea-8320-224f59045d34.png">

Editor-hat off, designer hat on: I think the underlines look distracting and not very pretty…  strong preference for no underline.

> 3. I think the currently active page/step should not be clickable in the widget. I think it should have a slightly different visual presentation, like in the menu bar.

I see the point, and it would be more correct, but I think it will make it less orderly visually, with no clear win (I would not expect users to click on the principle that they are currently on, and if they would, no-one would get hurt).

<img width="249" alt="results with current principle highlighted with white background" src="https://user-images.githubusercontent.com/178782/84383607-df95f600-abec-11ea-9573-34779a32cbba.png">

> 4. I think the text in this widget in the start/about page is not very useful. I think we should leave the same info here in the widget as we have on the other pages.

I think it is the best we can do on that page, and I feel it is helpful to those looking to understand what this page is for. Showing “Reported on X out of 39 success criteria.” is arguably less useful, because if you've not started yet, it is always going to be 0 out of X. Displaying that may feel like rushing people to evaluate more (something you've previously indicated)?

> 5. Should we an item in the widget that indicates whether start/about info has been provided? Seems also relevant for such an overview.

You mentioned this in the EOWG call… I don't support it, because:

* all info at the start is optional. Admittedly, reporting itself is _also_ optional, but reporting is the main point of the tool, so indication of progress seems most relevant there
* there are multiple items on the start page, indicating yes/no may not be useful enough, indicating each one would be overdoing it.
* by adding adding start/info, we increase the amount of information and take away prominence from the existing information. I don't think that's worth it.

If others feel strongly, let's discuss it tomorrow.

> 6. I don't feel too strongly but wonder if all fields/cells in this widget should be in just one column instead of currently two?

What would be the rationale of doing that? 

Advantage of two columns is that it saves space (half of it), and grabs less attention. I've designed the info to be compact enough to comfortably fit in half the space, this even works on the smaller end of mobile phones.

**Comment by nitedog** [comment]
2. I think some visual cue is necessary. I agree that the underline in this particular design is not very pretty. However, without underline, it is not clear to me that this is a link. Maybe find another design pattern that better communicates "link", and also one that you like as a designer?

3. I think such self-referential links are a problem for some people using screen magnification. For example, they only see that one part but do not see a change when they click. A change of style helps confirm visually that the click worked and the rest of the page has also changed.

PS: I think it is also best practice to remove the link functionality in addition to changing the visual style but I'm not absolutely sure about this -- would be good to check.

4. We have the same situation on the home/overview page. If we assume that people jump around the pages, I think this widget should maintain the same functionality. It seems odd to me to have a completely different function on only that particular page.

5. I think it is a useful reminder if people skip this at the start but also don't feel too strongly. Let's see what the rest of the group says.

6. I don't feel strongly at all about this but to me it seems fairly crammed and attention-grabbing as it currently is (I have to look and process where the correct cell is that I'm looking for -- in one line seems a little easier cognitively but admittedly I have not seen the 1-column layout).

7. (NEW) When the screen-size is small, and the widget drops to the bottom of the page -- should it come after the left/right navigation links (current implementation) or rather before? I'm unsure.

**Comment by hidde** [comment]
2. Why does it need to be clear that it is a link? On this GitHub page, none of the links look like links, yet people manage to navigate it. The chevron is my proposal to have a cue, an underline could also help `<designer-hat>`but is too much`</designer-hat>`. If we don't want either, I'd say we just leave it styled as it is.

> A change of style helps confirm visually that the click worked and the rest of the page

In our router, a new page focuses the page's `h1`, so that would even be reasonably clear when magnified. 

> If we assume that people jump around the pages, I think this widget should maintain the same functionality. It seems odd to me to have a completely different function on only that particular page.

If you feel strongly, could you open a new issue, so that it does not block the work here?

> admittedly I have not seen the 1-column layout

An example: 

<img width="286" alt="" src="https://user-images.githubusercontent.com/178782/84424496-7c25bb80-ac20-11ea-9972-c6dfabe8ffb7.png">

`<designer-hat>`I recommend against this, it takes much more space then needed.`</designer-hat>`.

**Comment by hidde** [comment]
I have updated the [preview](https://atag-report-tool-git-extend-eval-box.hdv.now.sh/principle/1), changes are:

* removed chevron from links to avoid confusion
* removed expand/collapse
* added real numbers to first item. These numbers are the evaluated items, not the items in A1, but it gives an idea of what this progress bar does while you evaluate as it reflects evaluations in real time – that is, when you leave the results selector).

**Comment by slhenry** [comment]
I like the cleanliness of this
![image](https://user-images.githubusercontent.com/6991814/84452999-01a76c80-ac1d-11ea-9849-3c58f4fb4946.png)

* **link indication** &mdash; I think it's fine that the A.1 etc don't have clickable affordance - because they are redundant with the links in the main navigation.
* **expand-collapse** &mdash; agree not to have the expand-collapse (and to be able to show/hide the whole box)
* **columns** &mdash; strong+1 for two columns. because they are so nicely parallel (4 As and 4 Bs) I think it looks really nice as two columns. I think it looks *more* complex as a single column. And I find it serves the purpose of giving an overview better as two columns.

> Should we an item in the widget that indicates whether start/about info has been provided? 

No. Unhelpful clutter.

> When the screen-size is small, and the widget drops to the bottom of the page -- should it come after the left/right navigation links (current implementation) or rather before? I'm unsure.

Mildly think it should be after, as it is now. (can explain rationale in meeting if anyone wants it :-)

<hr>

and a brainstorm: could also have a progress bar under "Reported on 0 out of 39 success criteria." going the full width. maybe nice, or maybe unnecessary clutter?

**Comment by hidde** [comment]
Screenshot for @slhenry brainstorm below, curious what people think! My thoughts: it could work and is a useful bit of information to access, but may also be too much. If we do it, probably important to make it look different: slightly larger font, slightly bigger, plenty of space in between.

<img width="288" alt="as preview, but with progress bar added underneath total" src="https://user-images.githubusercontent.com/178782/84490079-f139d500-aca2-11ea-95f1-ddaf908c3121.png">

**Comment by nitedog** [comment]
Overall, I really like this!

Some responses below:

> Why does it need to be clear that it is a link?

I consider that best practice and "in the spirit of WCAG". Here is *draft* guidance from the Cognitive and Learning Disabilities Task Force, which also goes in that direction:
 - https://www.w3.org/TR/coga-usable/#pattern-clearly-identify-controls-and-their-use

> I think it's fine that the A.1 etc don't have clickable affordance - because they are redundant with the links in the main navigation.

I could live with that but mild preference to clearer visual styling that this is a link.

> In our router, a new page focuses the page's h1, so that would even be reasonably clear when magnified.

I'm fine if you're confident it works (ie. doesn't distract by moving focus).

> If you feel strongly, could you open a new issue, so that it does not block the work here?

I don't feel strongly but opened another issue to not block this one.

> An example (<designer-hat>I recommend against this, it takes much more space then needed.</designer-hat>)

I actually like that 1-column layout but also acknowledge my lack of design skills.

**Comment by hidde** [comment]
Conclusions from discussion in EOWG (see comment above for IRC transcript):

To do: 

* [x] two columns
* [x] remove chevron, leave links as not specifically indicated (rationale: they are discoverable with mouse, keyboard, screenreader, and they redundant as navigation can also be done in the header or footer)
* [ ] add overall progress bar
* [ ] add option to minimise box, which updates the grid placement of the whole tool to make the content area wider, and take up content + sidebar size
* [ ] use real numbers for progress (currently hardcoded for demo)


Not to do: 

* Remove redundant links. The extra links to the Principles do get listed redundantly in screenreaders, but the benefits outweigh the downsides.
* Expand/collapse *within* the box. Not useful.

**Comment by hidde** [comment]
This now uses real numbers. Note, throughout the tool, when we say “X out of Y items”, we only count items within conformance target. So if your target is Level A and you evaluate an SC in Level AA, that does not count towards X or Y. This is to avoid you could get something like 40 out of 30.

**Comment by hidde** [comment]
Now ready: [prototype for optional evaluation box (hide/show button)](https://atag-report-tool-git-hide-sidebar.hdv.vercel.app/).

`<designer-hat>`My recommendation is against this, because: 

* full width view already available for users who zoom in / have a small device
* affects readability, as this lets too many words exist display per line
* hide/show button distracts from other functionality (have mitigated this somewhat by making it smaller and a less prominent colour)

First two could be mitigated by having the column used to the evaluation box remain empty, so that the content column continues to use the same width and does not become too wide.
`</designer-hat>`.

If the WG prefers, I can add this functionality.

**Comment by slhenry** [comment]
> Note, throughout the tool, when we say “X out of Y items”, we only count items within conformance target. 

Would be good to note that in the user documentation (on the fist page).

**Comment by slhenry** [comment]
> Now ready: prototype for optional evaluation box (hide/show button).

Thanks, Hidde! It really helps to be able to see it.

> hide/show button distracts from other functionality (have mitigated this somewhat by making it smaller and a less prominent colour)

How about like the Hide - Show Sidebar like in the Quick Ref (with the rounded borders of the new design).

> affects readability, as this lets too many words exist display per line

I support proving this functionality to give users more control of their own view.

For me, it's nice to be able to see more of the SC text and the Observations box at the same time. (as image below)
Also I note that ideally users of this tool know ATAG. They will be focusing on selecting the Results and typing in the Observations box.
I really like having the Observation boxes automatically go full width. That give me about 64 characters/line in the Observation box, which is perfect.

![image](https://user-images.githubusercontent.com/6991814/84670684-0ce3ec00-aeec-11ea-96b1-f2d19ce1b661.png)

**Comment by hidde** [comment]
> Thanks, Hidde! It really helps to be able to see it.

Yay! 

> How about like the Hide - Show Sidebar like in the Quick Ref (with the rounded borders of the new design).

I had something closer to it before:

<img width="256" alt="button with border and transparent background" src="https://user-images.githubusercontent.com/178782/84675061-06756480-af2c-11ea-866c-e6cd567baf0f.png">

but I found the border a bit too attention grabbing, I feel the dark gray background I've used now grabs less attention.

Or did you mean a chevron like in the QuickRef?

<img width="240" alt="Hide button with chevron pointing to the right" src="https://user-images.githubusercontent.com/178782/84675489-9b785d80-af2c-11ea-9dab-462b50a773d9.png">

>  I really like having the Observation boxes automatically go full width. That give me about 64 characters/line in the Observation box, which is perfect.

This is a nice benefit!

**Comment by slhenry** [comment]
> How about like the Hide - Show Sidebar like in the Quick Ref (with the rounded borders of the new design).

![image](https://user-images.githubusercontent.com/6991814/84675964-8bdc2300-aef2-11ea-969d-fe655e29585e.png)

with:
* chevron
* transparent background
* --line-grey border (which will have almost no contrast, which is OK 'cause of the chevron)


I think this will make it very subtle, not distracting, yet still clear.

**Comment by hidde** [comment]
<img width="287" alt="hide button with light grey border and chevron" src="https://user-images.githubusercontent.com/178782/84678144-f495c080-af2f-11ea-9ca7-9208050ed05e.png">

Like this?

**Comment by slhenry** [comment]
+1 from me.
What do you think?

**Comment by hidde** [comment]
I think it's the most subtle version yet! Will implement.

**Comment by nitedog** [comment]
Just to say that I really love this!

**Comment by bakkenb** [comment]
+1 on latest version.

**Comment by iadawn** [comment]
+1 from me
